test/e2e/multihoming.go

```diff
@@ -1458,3 +1423,53 @@ func kickstartPod(cs clientset.Interface, configuration podConfiguration) *v1.Po
 	}, 2*time.Minute, 6*time.Second).Should(Equal(v1.PodRunning))
 	return createdPod
 }
+
+func createNads(f *framework.Framework, nadClient nadclient.K8sCniCncfIoV1Interface, extraNamespace *corev1.Namespace, netConfig networkAttachmentConfig) error {
+	for _, ns := range []*corev1.Namespace{f.Namespace, extraNamespace} {
+		By(fmt.Sprintf("creating the nad for namespace %q", ns.Name))
+		netConfig.namespace = ns.Name
+		_, err := nadClient.NetworkAttachmentDefinitions(ns.Name).Create(
+			context.Background(),
+			generateNAD(netConfig),
+			metav1.CreateOptions{},
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	By("sitting on our hands for a couple secs we give the controller time to sync all NADs before provisioning policies and pods")
+	// TODO: this is temporary. We hope to eventually sync pods & multi-net policies on NAD C/U/D ops
+	time.Sleep(3 * time.Second)
+
+	return nil
+}
+
+func kickstartPodInNamespace(cs clientset.Interface, podConfig *podConfiguration, defaultNamespace string, extraNamespace string) *v1.Pod {
+	podConfig.namespace = getPodNamespace(defaultNamespace, podConfig.requiresExtraNamespace, extraNamespace)
+	return kickstartPod(cs, *podConfig)
+}
+
+func assertServerPodIPInRange(cidr string, serverIP string, netPrefixLengthPerNode int) {
+	By(fmt.Sprintf("asserting the server pod IP %v is from the configured range %v/%v", serverIP, cidr, netPrefixLengthPerNode))
+	subnet, err := getNetCIDRSubnet(cidr)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+	ExpectWithOffset(1, inRange(subnet, serverIP)).To(Succeed())
+}
+
+func createMultiNetworkPolicy(mnpClient mnpclient.K8sCniCncfIoV1beta1Interface, namespace string, policy *mnpapi.MultiNetworkPolicy) error {
+	By("provisioning the multi-network policy")
+	_, err := mnpClient.MultiNetworkPolicies(namespace).Create(
+		context.Background(),
+		policy,
+		metav1.CreateOptions{},
+	)
+	return err
+}
+
+func getPodNamespace(defaultNamespace string, requiresExtraNamespace bool, extraNamespace string) string {
```

Comment:
nit: I'd just do this is the caller. This function's signature is IMHO very weird. It is called get pod namespace, does not receive a pod - but the namespaces - plus a boolean.